source/views/directory/list.js

```diff
@@ -0,0 +1,244 @@
+// @flow
+
+import React from 'react'
+import {StyleSheet, View, Text, Image, Platform} from 'react-native'
+import {SearchableAlphabetListView} from '@frogpond/listview'
+import debounce from 'lodash/debounce'
+import type {TopLevelViewPropsType} from '../types'
+import {Row, Column} from '@frogpond/layout'
+import {
+	ListRow,
+	ListSectionHeader,
+	ListSeparator,
+	Detail,
+	Title,
+} from '@frogpond/lists'
+import size from 'lodash/size'
+import sortBy from 'lodash/sortBy'
+import groupBy from 'lodash/groupBy'
+import head from 'lodash/head'
+import uniq from 'lodash/uniq'
+import words from 'lodash/words'
+import deburr from 'lodash/deburr'
+import filter from 'lodash/filter'
+import isString from 'lodash/isString'
+import {fetch} from '@frogpond/fetch'
+import * as c from '@frogpond/colors'
+import startCase from 'lodash/startCase'
+import type {DirectoryType} from './types'
+import Icon from 'react-native-vector-icons/Ionicons'
+import isEmpty from 'lodash/isEmpty'
+
+const url = 'https://www.stolaf.edu/directory/index.cfm'
+
+const leftSideSpacing = 20
+const rowHeight = Platform.OS === 'ios' ? 58 : 74
+const headerHeight = Platform.OS === 'ios' ? 33 : 41
+
+const styles = StyleSheet.create({
+	wrapper: {
+		flex: 1,
+	},
+	row: {
+		height: rowHeight,
+		marginLeft: 22,
+	},
+	rowSectionHeader: {
+		height: headerHeight,
+	},
+	image: {
+		width: 35,
+		marginRight: 10,
+	},
+	emptySearch: {
+		flex: 1,
+		top: -150,
+		alignItems: 'center',
+	},
+	emptySearchText: {
+		fontSize: 18,
+		color: c.black,
+		textAlign: 'center',
+		paddingTop: 20,
+		paddingBottom: 10,
+	},
+})
+
+export class DirectoryView extends React.Component {
+	static navigationOptions = {
+		title: 'Directory',
+		headerBackTitle: 'Home',
+	}
+
+	props: TopLevelViewPropsType
+
+	searchBar: any
+
+	state: {
+		results: {[key: string]: Array<DirectoryType>},
+		refreshing: boolean,
+		error: boolean,
+		loaded: boolean,
+	} = {
+		results: {},
+		refreshing: false,
+		error: false,
+		loaded: false,
+	}
+
+	fetchData = async (query: string) => {
+		try {
+			let responseData: Array<DirectoryType> = await fetch(url, {
+				cache: 'no-store',
+				searchParams: {
+					fuseaction: 'SearchResults',
+					format: 'json',
+					query: query,
+				},
+			}).json()
+
+			const results = responseData.results
+
+			const sortableRegex = /^(St\.? Olaf(?: College)?|The) +/iu
+			const withSortableNames = results.map(item => {
+				const sortableName = item.lastName.replace(sortableRegex, '')
+
+				return {
+					...item,
+					$sortableName: sortableName,
+					$groupableName: head(startCase(sortableName)),
+				}
+			})
+
+			const sorted = sortBy(withSortableNames, '$sortableName')
+			const grouped = groupBy(sorted, '$groupableName')
+			this.setState(() => {
+				results: grouped
```

Comment:
This is not returning an object; it's returning `undefined`, after evaluating a block with a `results:` label. You will need to wrap the {…} in parentheses in order for it to be parsed as an object.